components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -429,30 +434,35 @@ public class NetworkMessageEncoder : IMessageEncoder {
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (_knownPayloadIdentifiers.TryGetValue(notification.NodeId.ToString(), out var knownPayloadIdentifier)) {
-                if (!string.IsNullOrEmpty(knownPayloadIdentifier)) {
-                    return knownPayloadIdentifier;
-                }
+            if (_knownPayloadIdentifiers.TryGetValue(notification.NodeId.ToString(), out var knownPayloadIdentifier) && !string.IsNullOrEmpty(knownPayloadIdentifier)) {
```

Comment:
>notification.NodeId.ToString() [](http://example.com/codeflow?start=53&length=30)

Avoid multiple calls to notification.NodeId.ToString(), store it as local variable

Reply:
Fixed.